qds_sdk/clusterv2.py

```diff
@@ -0,0 +1,549 @@
+                                      custom_tags=arguments.custom_tags,
+                                      heterogeneous_config=arguments.heterogeneous_config,
+                                      maximum_bid_price_percentage=arguments.maximum_bid_price_percentage,
+                                      timeout_for_request=arguments.timeout_for_request,
+                                      maximum_spot_instance_percentage=arguments.maximum_spot_instance_percentage,
+                                      stable_maximum_bid_price_percentage=arguments.stable_maximum_bid_price_percentage,
+                                      stable_timeout_for_request=arguments.stable_timeout_for_request,
+                                      stable_spot_fallback=arguments.stable_spot_fallback,
+                                      idle_cluster_timeout=arguments.idle_cluster_timeout,
+                                      disk_count=arguments.count,
+                                      disk_type=arguments.disk_type,
+                                      disk_size=arguments.size,
+                                      upscaling_config=arguments.upscaling_config,
+                                      enable_encryption=arguments.encrypted_ephemerals,
+                                      customer_ssh_key=customer_ssh_key)
+
+        #  This will set cloud config settings
+        cloud_config = Cloud.get_cloud()
+        cloud_config.set_cloud_config_from_arguments(arguments)
+
+        # This will set engine settings
+        engine_config = Engine(flavour=arguments.flavour)
+        engine_config.set_engine_config_settings(arguments)
+
+        cluster_request = ClusterCmdLine.get_cluster_request_parameters(cluster_info, cloud_config, engine_config)
+
+        action = args[0]
+        if action == "create":
+            return arguments.func(cluster_request)
+        else:
+            return arguments.func(arguments.cluster_id_label, cluster_request)
+
+    @staticmethod
+    def get_cluster_request_parameters(cluster_info, cloud_config, engine_config):
+
+        '''
+        Use this to return final minimal request from cluster_info, cloud_config or engine_config objects
+        Alternatively call util._make_minimal if only one object needs to be implemented
+        '''
+
+        cluster_request = {}
+        cloud_config = util._make_minimal(cloud_config.__dict__)
+        if bool(cloud_config): cluster_request['cloud_config'] = cloud_config
+
+        engine_config = util._make_minimal(engine_config.__dict__)
+        if bool(engine_config): cluster_request['engine_config'] = engine_config
+
+        cluster_request.update(util._make_minimal(cluster_info.__dict__))
+        return cluster_request
+
+class ClusterInfoV2(object):
+    """
+    qds_sdk.ClusterInfoV2 is the class which stores information about a cluster_info.
+    You can use objects of this class to create/update/clone a cluster.
+    """
+
+    def __init__(self, label):
```

Comment:
For old api v1.2 and v1.3 we are doing like that only. It is because label is the primary and mandatory parameter for cluster apis call while creating , updating(we can use label instead of id) and cloning cluster. It will also help customers that definition is same and use it as same like we have in old cluster apis.